src/NServiceBus.RavenDB/Outbox/OutboxPersister.cs

```diff
@@ -36,10 +34,15 @@ public OutboxPersister(IDocumentStore documentStore, EndpointName endpointName)
                 return default(OutboxMessage);
             }
 
-            var operations = new List<TransportOperation>();
+            var operations = new TransportOperation[result.TransportOperations.Length];
             if (!result.Dispatched)
             {
-                operations.AddRange(result.TransportOperations.Select(t => new TransportOperation(t.MessageId, t.Options, t.Message, t.Headers)));
+                var index = 0;
+                foreach (var operation in result.TransportOperations)
```

Comment:
I tried to add that in a comment and couldn't make it not sound dumb. It should be obvious that arrays are being used instead of LINQ for perf. `foreach` only shows up on our radar during this PR because we were in the specific mindset of perf. I don't think anyone would look at this and try to re-LINQ it or try to change the `foreach` to `for` just for fun.